controller/api/public/http_server_test.go

```diff
@@ -0,0 +1,208 @@
+package public
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	common "github.com/runconduit/conduit/controller/gen/common"
+	healcheckPb "github.com/runconduit/conduit/controller/gen/common/healthcheck"
+	pb "github.com/runconduit/conduit/controller/gen/public"
+)
+
+type mockGrpcServer struct {
+	LastRequestReceived proto.Message
+	ResponseToReturn    proto.Message
+	TapStreamsToReturn  []*common.TapEvent
+	ErrorToReturn       error
+}
+
+func (m *mockGrpcServer) Stat(ctx context.Context, req *pb.MetricRequest) (*pb.MetricResponse, error) {
+	m.LastRequestReceived = req
+	return m.ResponseToReturn.(*pb.MetricResponse), m.ErrorToReturn
+}
+
+func (m *mockGrpcServer) Version(ctx context.Context, req *pb.Empty) (*pb.VersionInfo, error) {
+	m.LastRequestReceived = req
+	return m.ResponseToReturn.(*pb.VersionInfo), m.ErrorToReturn
+}
+
+func (m *mockGrpcServer) ListPods(ctx context.Context, req *pb.Empty) (*pb.ListPodsResponse, error) {
+	m.LastRequestReceived = req
+	return m.ResponseToReturn.(*pb.ListPodsResponse), m.ErrorToReturn
+}
+
+func (m *mockGrpcServer) SelfCheck(ctx context.Context, req *healcheckPb.SelfCheckRequest) (*healcheckPb.SelfCheckResponse, error) {
+	m.LastRequestReceived = req
+	return m.ResponseToReturn.(*healcheckPb.SelfCheckResponse), m.ErrorToReturn
+}
+
+func (m *mockGrpcServer) Tap(req *pb.TapRequest, tapServer pb.Api_TapServer) error {
+	m.LastRequestReceived = req
+	if m.ErrorToReturn == nil {
+		for _, msg := range m.TapStreamsToReturn {
+			tapServer.Send(msg)
+		}
+	}
+
+	return m.ErrorToReturn
+}
+
+type grpcCallTestCase struct {
+	expectedRequest  proto.Message
+	expectedResponse proto.Message
+	functionCall     func() (proto.Message, error)
+}
+
+func TestServer(t *testing.T) {
+
+	mockGrpcServer := &mockGrpcServer{}
+	handler := &handler{
+		grpcServer: mockGrpcServer,
+	}
+
+	httpServer := &http.Server{
+		Addr:    "localhost:8889",
+		Handler: handler,
+	}
+
+	go func() {
+		err := httpServer.ListenAndServe()
+		if err != nil {
+			fmt.Println(err)
```

Comment:
I don't think this will show up in standard `go test` output. I recommend using `t.Log(err)` instead (or, alternatively, removing in since the error is logged in the next line anyway).

Reply:
It will with `-v` but tbh I'm pretty sure this is vestigial from me debugging some problems I had running these tests on travis so I'll remove it.